core/serial_ftdi.c

```diff
@@ -467,7 +467,7 @@ static dc_status_t serial_ftdi_write (void **userdata, const void *data, size_t
 
 		// Calculate the expected duration (microseconds). A 2 millisecond fudge
 		// factor is added because it improves the success rate significantly.
-		unsigned long expected = 1000000.0 * device->nbits / device->baudrate * size + 0.5 + 2000;
+		unsigned long expected = 1000000 * device->nbits / device->baudrate * size + 2000;
```

Comment:
assuming that `nbits` and `badrate` are reasonable it can overflow the 32bit `long` only if `size` is a very big value (e.g. >5MB). `unsigned long long` is an option for c99.